### PR TITLE
Add setCollateralToken rpc

### PR DESCRIPTION
**Pull request by jingyi2811**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Add setCollateralToken rpc

#### Which issue(s) does this PR fixes?:

Fixes part of #488

### CODE REVIEW

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 1]
```diff
@@ -26,6 +26,23 @@ export class Loan {
...
   }
+
+  /**
+   * Set a collateral token transaction.
+   *
+   * @param {SetCollateralToken} collateralToken
+   * @param {string} collateralToken.token Symbol or id of collateral token
+   * @param {BigNumber} collateralToken.factor Collateralization factor
+   * @param {string} collateralToken.priceFeedId txid of oracle feeding the price
+   * @param {number} [collateralToken.activateAfterBlock] changes will be active after the block height (Optional)
```
Comment (Jouzo) [inline packages/jellyfish-api-core/src/category/loan.ts:37]:
```suggestion
   * @param {number} [collateralToken.activateAfterBlock] changes will be active after the block height
```
Redundant (optional) as [] already means that it's an optional parameter

**Review by Jouzo — APPROVED** [review]

**Review by canonbrother — DISMISSED** [review]

**Review by surangap — DISMISSED** [review]

File: packages/jellyfish-api-core/src/category/loan.ts (hunk 1 above)
Comment (surangap) [reply packages/jellyfish-api-core/src/category/loan.ts:74]: wondering what happened to this😉

Comment (jingyi2811) [reply packages/jellyfish-api-core/src/category/loan.ts:74]: Forgot to remove. Just removed.  Thanks.

**Review by Jouzo — APPROVED** [review]

File: packages/jellyfish-api-core/__tests__/category/loan/setCollateralToken.test.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    const promise = testing.rpc.loan.setCollateralToken({
+      token: 'AAPL',
+      factor: new BigNumber(0.5),
+      priceFeedId
+    }, [utxo])
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSetCollateralTokenTx execution failed:\ntx not from foundation member!\', code: -32600, method: setcollateraltoken')
+  })
+})
+
+describe('Loan with activateAfterBlock at block 160', () => {
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/setCollateralToken.test.ts:112]:
```suggestion
describe('setCollateralToken with activateAfterBlock', () => {
```

File: packages/jellyfish-api-core/__tests__/category/loan/setCollateralToken.test.ts [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        token: 'AAPL',
+        factor: 0.5,
+        priceFeedId,
+        activateAfterBlock: 120
+      }
+    })
+  })
+})
+
+describe('Loan with activateAfterBlock less than current block', () => {
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/setCollateralToken.test.ts:161]:
```suggestion
describe('setCollateralToken with activateAfterBlock less than current block', () => {
```

File: packages/jellyfish-api-core/__tests__/category/loan/setCollateralToken.test.ts [hunk 4]
```diff
@@ -0,0 +1,196 @@
+import { LoanMasterNodeRegTestContainer } from './loan_container'
+import BigNumber from 'bignumber.js'
+import { Testing } from '@defichain/jellyfish-testing'
+import { GenesisKeys } from '@defichain/testcontainers'
+
+describe('Loan', () => {
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/setCollateralToken.test.ts:6]:
```suggestion
describe('Loan setCollateralToken', () => {
```

**Review by surangap — APPROVED** [review]

**Review by Jouzo — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**